docs/Documentation/Development/Languages/Python/dask.md

```diff
@@ -9,172 +9,192 @@ grand_parent: Programming Languages
-
-```python
-import dask
-import time
-from dask_jobqueue import SLURMCluster
-from distributed import Client
-import random 
-
-@dask.delayed
-def inc(x):
-    time.sleep(random.random())
-    return x + 1
-
-@dask.delayed
-def dec(x):
-    time.sleep(random.random())
-    return x - 1
-
-@dask.delayed
-def add(x, y):
-    time.sleep(random.random())
-    return x + y
-
-def main ():
-   cluster = SLURMCluster(
-      cores=18,
-      memory='24GB',
-      queue='short',
-      project='hpcapps',
-      walltime='00:30:00',
-      interface='ib0',
-      processes=18,
-   )
-   cluster.scale(jobs=2)
-
-   client = Client(cluster)
-   zs = []
-   for i in range(256):
-      x = inc(i)
-      y = dec(x)
-      z = add(x, y)
-      zs.append(z)
-   
-  
-   result = dask.compute(*zs)
-   print (result)
-
-
-if __name__ == "__main__":
-   main()
+Note that although 2 jobs are requested, Dask launches the jobs dynamically, so depending on the status of the job queue, your results may indicate that only a single node was used.
+
 
+## Dask MPI
+
+Dask also provides a package called [`dask-mpi`](http://mpi.dask.org/en/latest/index.html) that uses MPI to create the cluster.  Note that `dask-mpi` only uses MPI to start the cluster, not for inter-node communication.
+
+Dask-MPI provides two interfaces to launch Dask, either from a batch script using the Python API, or from the command line.
+
+Here we show a simple example that uses Dask-MPI with a batch script.  Make sure that you have installed `dask-mpi` following the [Installation Instructions](#installation).  Create `dask_mpi_example.py` and `dask_mpi_launcher.sh` with the contents below.  In `dask_mpi_launcher.sh`, replace `<project>` with your allocation.
+
+??? example "`dask_mpi_example.py`"
+
+    ```python
+    from dask_mpi import initialize
+    from dask.distributed import Client
+    import socket
+    import time
+    from collections import Counter
+    
+    def test():
+       return socket.gethostname()
+       
+    def main():
+       initialize(interface='ib0',
+                  nthreads=5)
+       client = Client()
+       time.sleep(15)
+    
+       result = []
+    
+       for i in range (0,100):
+          result.append(client.submit(test).result())
+          time.sleep(1)
+          
+       out = str(Counter(result))
+       print(f'nodes: {out}')
+    
+    main()
+    ```
+    
+??? example "`dask_mpi_launcher.sh`"
+
+    ```bash
+    #!/bin/bash 
+    #SBATCH --nodes=2
+    #SBATCH --ntasks=4
+    #SBATCH --time=10
+    #SBATCH --account=<project>
+    #SBATCH --partition=debug
+    
+    ml intel-mpi
+    mpiexec -np 4 python dask_mpi_example.py
+    ```
+    
+The job is then launched as:
+
+```bash
+sbatch dask_mpi_launcher.sh
```

Comment:
An example combining `dask-jobqueue` and `dask-mpi` on the HPC could be useful. That would help give users some context as to when to use which tool and how to combine them.

Reply:
I haven't used `SLURMCluster` but the changes needed to run multi-node jobs seem about the same as those needed for dask-mpi. Digging deeper,  I agree that `dask-jobqueue` and `dask-mpi` cannot be combined in a single run. 

I did have a question though, what happens when you run `python dask_slurm_example.py` and you log out of the login node. Does the job continue to run while the user is "offline"?

Reply:
If you need to log out you can use tmux or nohup to background the process.  But if you kill the main Python process, that will kill the scheduler.  It is supposed to send a signal to stop the workers, and apparently they also kill themselves automatically after 60 seconds with a non-existent scheduler.